### PR TITLE
Action output

### PR DESCRIPTION
**Pull request by gabriellemacinnes**
#773 
Added the actions of people detected in the graphic to TTS. If there are multiple people in the graphic the output contains the total number of people followed by the actions each are doing, grouped based on type of action and confidence. The confidence threshold is set to 0.85, if the confidence for the action of a certain person is below this threshold then the output says they might be doing this action. These changes were tested by using the test extension to interpret graphics with one person doing an action, graphics with several people doing the same action, graphics with several people doing various actions, and graphics with no people.
Examples:
image
Text output: This photo contains the following outlines of regions: sky, and mountain. It also contains the following objects or people: a person climbing.

image
Text output: This outdoor photo contains the following outlines of regions: person, tree, grass, and wall. It also contains the following objects or people: 7 people doing the following actions: 3 people jumping, 2 people who might be fighting, and 2 other people.

image
Text output: This photo contains the following outlines of regions: person, sky, tree, wall, and fence. It also contains the following objects or people: a sports ball, and 3 people doing the following actions: 1 person jumping, 1 person who might be fighting, and 1 other person.

---

## Required Information

- [x] I referenced the issue addressed in this PR.
- [x] I described the changes made and how these address the issue.
- [x] I described how I tested these changes.

## Coding/Commit Requirements

* [x] I followed applicable coding standards where appropriate (e.g., [PEP8](https://pep8.org/))
* [x] I have not committed any models or other large files.

## New Component Checklist (**mandatory** for new microservices)

* [ ] I added an entry to `docker-compose.yml` and `build.yml`.
* [ ] I created A CI workflow under `.github/workflows`.
* [ ] I have created a `README.md` file that describes what the component does and what it depends on (other microservices, ML models, etc.).

OR
* [ ] I have not added a new component in this PR.

### CODE REVIEW

File: handlers/photo-audio-handler/src/utils.ts [hunk 1]
```diff
@@ -114,38 +118,127 @@ export function filterObjectsBySize(objDet: ObjDet, objGroup: ObjGroup) {
...
+        objects.push(object);
+    }
+
+    return objects;
+}
 
-export function generateObjDet(objDet: ObjDet, objGroup: ObjGroup): TTSSegment[] {
+export function generateObjDet(objDet: ObjDet, objGroup: ObjGroup, actionRec: Action): TTSSegment[] {
+    let actionOut = false;
+    if (actionRec && actionRec["actions"].length > 0) { actionOut = true }
```
Comment (JRegimbal) [inline handlers/photo-audio-handler/src/utils.ts:192]: Combine these lines.

File: handlers/photo-audio-handler/src/utils.ts [hunk 2]
```diff
@@ -114,38 +118,127 @@ export function filterObjectsBySize(objDet: ObjDet, objGroup: ObjGroup) {
...
     const objects: TTSSegment[] = [];
     objects.push({"type": "text", "value": "contains the following objects or people:"});
+    let actionTTS: TTSSegment[] = [];
```
Comment (JRegimbal) [inline handlers/photo-audio-handler/src/utils.ts:195]: Please add comments to this section.

**Review by JRegimbal — CHANGES_REQUESTED** [review]
A few minor things, but this is a great pull request. Good work!

**Comment by jeffbl** [comment]
@gabriellemacinnes Note that one of the tests for this should be to run the modified photo-audio handler both with and without the activity recognition preprocessor running, so that end-users get results as before when activity recognition preprocessor is not running, but it includes the results when it is running.

**Comment by gabriellemacinnes** [comment]
@jeffbl Sample output when action preprocessor is running:
This outdoor photo contains the following outlines of regions: earth, grass, person, sky, and tree. It also contains the following objects or people: a person running.
Output on the same image after stopping action container:
This outdoor photo contains the following outlines of regions: earth, grass, person, sky, and tree. It also contains the following objects or people: a person.
For reference this is the image:
image

**Comment by gabriellemacinnes** [comment]
@JRegimbal I replaced underscores in labels with spaces because I just realized the tts of actions with underscores wasn't very good.

**Comment by JRegimbal** [comment]
Looks good to me - well done!

**Review by JRegimbal — APPROVED** [review]

**Pull request merged**